Dockerfile

```
@@ -0,0 +1,8 @@
+FROM node:20-alpine
+USER node
+WORKDIR /usr/local/app
+COPY package.json .
+RUN yarn install --ignore-scripts
+EXPOSE ${APP_PORT}
+COPY . .
```

Since you have a proper `.dockerignore` I would recommend you ignore this line from Sonarqube's analysis. Otherwise it's going to keep complaining.

https://stackoverflow.com/a/30383335

```suggestion
COPY . . # NO SONAR
```

---

Also about the # NO SONAR property, sometimes it refuses to work so you can also try this approach @Pratham Sapkota 

Reference `sonar-project`: https://github.com/templ-project/typescript-nestjs/blob/master/sonar-project.properties
Excluding a sonar error: https://stackoverflow.com/questions/21323276/sonarqube-exclude-a-directory